src/Generator/Passes/CleanCommentsPass.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using CppSharp.AST;
+using CppSharp.Generators.CSharp;
+using System.Text.RegularExpressions;
+
+namespace CppSharp.Passes
+{
+    public class CleanCommentsPass : TranslationUnitPass,ICommentVisitor<bool>
+    {
+        public bool VisitBlockCommand(BlockCommandComment comment)
+        {
+            return true;
+        }
+
+        public override bool VisitDeclaration(Declaration decl)
+        {
+            if (!base.VisitDeclaration(decl))
+                return false;
+
+            if (decl.Comment != null)
+            {
+                var fullComment = decl.Comment.FullComment;
+                VisitFull(fullComment);
+            }
+            return true;
+        }
+
+        public bool VisitFull(FullComment comment)
+        {
+            foreach (var block in comment.Blocks)
+                block.Visit(this);
+
+            return true;
+        }
+        #region Comments Visit
+        public bool VisitHTMLEndTag(HTMLEndTagComment comment)
+        {
+            return true;
+        }
+
+        public bool VisitHTMLStartTag(HTMLStartTagComment comment)
+        {
+            return true;
+        }
+
+        public bool VisitInlineCommand(InlineCommandComment comment)
+        {
+            return true;
+        }
+
+        public bool VisitParagraphCommand(ParagraphComment comment)
+        {
+            bool tag = false;
+            foreach (var item in comment.Content.Where(c=> c.Kind == DocumentationCommentKind.TextComment))
+            {
+                TextComment com = (TextComment) item;
+                if (!Generators.Helpers.RegexTag.IsMatch(com.Text) && tag)
```

Comment:
Since I am pretty sure the correct path now that you have an else if the old one - if (regex) { if (tag) } invert the if so that you have a positive condition.